src/core/thread/network_data_leader_ftd.cpp

```diff
@@ -291,20 +291,22 @@ void Leader::HandleCommissioningGet(Coap::Message &aMessage, const Ip6::MessageI
     uint8_t      tlvs[NetworkData::kMaxSize];
     uint8_t      length = 0;
 
-    while (offset < aMessage.GetLength())
+    while (offset + sizeof(tlv) <= aMessage.GetLength())
```

Comment:
How about using `Tlv::GetValueOffset()` to find the value offset and length for TLV type `MeshCoP::Tlv::kGet` here?

Reply:
Just submitted #4194 to replace this PR.